### PR TITLE
ART-15994: fix(images): implement golang builder base image workflow

### PR DESCRIPTION
**Pull request by lgarciaaco**
## Summary
Implements comprehensive golang builder base image workflow to resolve [ART-15994](https://redhat.atlassian.net/browse/ART-15994). This change enables golang builder images to properly trigger the base image release workflow with correct component naming and Jenkins integration. The implementation includes enhanced detection logic, centralized component naming, and robust test coverage.

## Dependencies
**⚠️ Merge Dependencies - Required in order**:
1. **Jenkins Job Update**: https://github.com/openshift-eng/aos-cd-jobs/pull/4658 - Updates base-image-release job to accept golang builder group naming conventions
2. **Konflux Release Data**: https://gitlab.cee.redhat.com/releng/konflux-release-data/-/merge_requests/17055 - Required Konflux configuration changes

Both dependencies must be merged and deployed before this PR can be safely merged.

## Problem
**Before:** 
- Golang builder images failed to trigger base image release workflow due to naming incompatibility
- Component names used full semantic versions (golang-builder-v1.25.8-rhel8) instead of simplified format
- Jenkins BUILD_VERSION parameter only accepted OCP format, causing HTTP 500 errors for golang builders
- Inline logic scattered across base_image_handler.py with no centralized component naming

**After:** 
- Golang builder images properly detected and trigger base image release workflow
- Component names use simplified major.minor format (golang-builder-v1.25-rhel8) 
- Jenkins integration works with both OCP and golang builder group naming
- Centralized component naming logic in KonfluxImageBuilder with comprehensive test coverage

### Key Changes
- **Enhanced golang builder detection**: Support both `openshift-golang-builder` and `openshift/golang-builder` name formats in `image.py:1320`
- **Centralized component naming**: Added `KonfluxImageBuilder.get_golang_builder_component_name()` static method with NVR parsing and version extraction
- **Simplified version format**: Extract major.minor only from semantic versions (v1.25.8 → v1.25) for component names
- **Jenkins integration**: Added job title/description updates with component information for better visibility
- **Comprehensive testing**: 8 test methods covering 50+ real-world NVR examples with edge cases
- **Removed environment validation**: Cleaned up debugging logic for production deployment
- **Fixed build version passing**: Use full group name instead of extracted version for Jenkins calls

### Component Name Format
`golang-builder-v{major.minor}-{rhel_version}`

Examples:
- `openshift-golang-builder-container-v1.25.8-202604081607.p0.g2aa6a05.el8` → `golang-builder-v1.25-rhel8`
- `openshift-golang-builder-container-v1.24.13-202603271102.p2.ge8e5642.el9` → `golang-builder-v1.24-rhel9`

### Files Modified
- `doozer/doozerlib/backend/base_image_handler.py`: Refactored component naming logic, added Jenkins updates
- `doozer/doozerlib/backend/konflux_image_builder.py`: Added golang builder component naming method  
- `doozer/doozerlib/image.py`: Enhanced golang builder detection for both name formats
- `doozer/tests/backend/test_konflux_image_builder.py`: Comprehensive test suite with real NVR dataset

## Jira
- **Ticket**: [ART-15994](https://redhat.atlassian.net/browse/ART-15994)
- **Priority**: Major
- **Status**: In Progress

### CODE REVIEW

**Comment by lgarciaaco** [comment]
## 🧪 Testing Update - Golang Builder Workflow Validation

Successfully tested the golang builder base image workflow end-to-end with the changes in this PR:

### Test Flow Execution
1. **Golang Builder Job**: [golang-builder/26](https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/hack/job/lgarciaa/job/build%252Fgolang-builder/26/console) - ✅ Successfully triggered
2. **Base Image Release Job**: [build/base-image-release/24](https://art-jenkins.apps.prod-stable-spoke1-dc-iad2.itup.redhat.com/job/hack/job/lgarciaa/job/build%252Fbase-image-release/24/console) - ✅ Successfully triggered by golang builder
3. **PipelineRun Created**: [managed-ddrfc](https://konflux-ui.apps.kflux-ocp-p01.7ayg.p1.openshiftapps.com/ns/rhtap-releng-tenant/applications/art-images-base/pipelineruns/managed-ddrfc?releaseName=ocp-base-image-release-sdxm9) - ⏳ Created successfully, pending dependency

### Results
- **Component naming** ✅: Golang builder component names generated correctly with major.minor format 
- **Jenkins integration** ✅: Base image release job properly triggered by golang builder workflow
- **PipelineRun generation** ✅: PLR created successfully in Konflux
- **Expected failure**: PLR failed due to **missing Konflux release data dependency** (MR 17055)

### Validation Reference
Identical successful PLR for same golang builder after dependencies: [ocp-base-image-release-pspb9](https://konflux-ui.apps.kflux-ocp-p01.7ayg.p1.openshiftapps.com/ns/ocp-art-tenant/applications/art-images-base/releases/ocp-base-image-release-pspb9)

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 1]
```diff
@@ -510,6 +510,43 @@ def get_component_name(application_name: str, image_name: str):
...
+        else:
+            version_parts = version.split(".")
+
+        if len(version_parts) >= 2:
+            major_minor = f"v{version_parts[0]}.{version_parts[1]}"
+        else:
+            # Fallback to original version if parsing fails
+            major_minor = version
+
+        # Determine RHEL version from release field
```
Comment (fgallott) [inline doozer/doozerlib/backend/konflux_image_builder.py:540]: what about something like

```suggestion
        # Determine RHEL version from release field
        el_suffix = "rhel9"
        if ".el8" in release:
            el_suffix = "rhel8"

```

**Review by fgallott — COMMENTED** [review]
a simple suggestion but 
/lgtm
